core/src/banking_stage/transaction_scheduler/prio_graph_scheduler.rs

```diff
@@ -68,6 +69,23 @@ impl PrioGraphScheduler {
         pre_lock_filter: impl Fn(&SanitizedTransaction) -> bool,
     ) -> Result<SchedulingSummary, SchedulerError> {
         let num_threads = self.consume_work_senders.len();
+        let max_cu_per_thread = MAX_BLOCK_UNITS / num_threads as u64;
```

Comment:
Just to clarify: assume there is `4` work threats, so `max_cu_per_thread` could allow 8.5 TXs (each has 1.4M CU) in flight. In another words, `num_threads` must lesser than `34`, otherwise transactions with 1.4M Cu could not be scheduled.

Reply:
probably poor naming on my part. It is a "soft" limit. we will schedule 1 tx over the limit, we only remove from the schedulable set if we are at or above the limit.
So if we had 4 threads each with `11_999_999` in-flight. We could still schedule a 1.4m for the implicit hard cap of `13_399_999`.